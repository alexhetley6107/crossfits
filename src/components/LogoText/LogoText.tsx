import { theme } from '@/themes';
import React from 'react';
import styled from 'styled-components';

type SizeUnion = 'sm' | 'lg';

export type LogoTextProps = {
  className?: string;
  size?: SizeUnion;
};

const sizeMap: Record<SizeUnion, string> = {
  sm: '24px',
  lg: '32px',
};

const LogoTextWrapper = styled.div<{ $size: SizeUnion }>`
  margin: 0;
  padding: 0;
  font-size: ${({ $size }) => sizeMap[$size]};
  font-weight: 600;
  font-style: italic;
  color: ${theme.colors.black};
  span {
    color: ${theme.colors.orange};
  }
`;

export const LogoText: React.FC<LogoTextProps> = ({ size = 'sm', ...props }) => {
  return (
    <LogoTextWrapper data-testid="LogoText" $size={size} {...props}>
      Remin<span>Dee</span>
    </LogoTextWrapper>
  );
};
