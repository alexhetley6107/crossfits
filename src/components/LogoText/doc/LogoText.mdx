import { Canvas, Meta } from '@storybook/blocks';
import { Example } from './Example.tsx';
import * as LogoText from './LogoText.stories.tsx';

<Meta of={LogoText} title="LogoText" />

# LogoText

<Canvas of={LogoText.Base} />

## Usage

```ts
import { LogoText } from 'remindee-ui-kit';

export const Example = () => {
  return <LogoText />;
};
```

### Arguments

- **className** - A string that sets css class to the root of component.