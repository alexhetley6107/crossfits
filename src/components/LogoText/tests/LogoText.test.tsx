import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { LogoText } from '../LogoText';
import '@testing-library/jest-dom';
import 'jest-styled-components';

describe('LogoText component', () => {
  describe('General', () => {
    it('Render', () => {
      render(<LogoText />);
      const el = screen.getByTestId('LogoText');
      expect(el).toBeInTheDocument();
    });
  });
  describe('Size', () => {
    it('Small', () => {
      render(<LogoText size="sm" />);
      const el = screen.getByTestId('LogoText');
      expect(el).toMatchSnapshot();
    });
    it('Large', () => {
      render(<LogoText size="lg" />);
      const el = screen.getByTestId('LogoText');
      expect(el).toMatchSnapshot();
    });
  });
});
