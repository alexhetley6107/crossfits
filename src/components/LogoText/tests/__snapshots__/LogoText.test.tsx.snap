// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogoText component > Size > Large 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  font-size: 32px;
  font-weight: 600;
  font-style: italic;
  color: #333;
}

.c0 span {
  color: #ff7705;
}

<div
  class="c0"
  data-testid="LogoText"
>
  Remin
  <span>
    Dee
  </span>
</div>
`;

exports[`LogoText component > Size > Small 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  font-weight: 600;
  font-style: italic;
  color: #333;
}

.c0 span {
  color: #ff7705;
}

<div
  class="c0"
  data-testid="LogoText"
>
  Remin
  <span>
    Dee
  </span>
</div>
`;
